snow/engine/common/tracker/peers.go

```diff
@@ -37,10 +36,10 @@ type Peers interface {
 	SampleValidator() (ids.NodeID, bool)
 	// GetValidators returns the set of all validators
 	// known to this peer manager
-	GetValidators() set.Set[ids.NodeID]
+	GetValidators() set.Set[ids.NodeWeight]
```

Comment:
Would it make sense to use the validator set type on the snowman config type rather than making this change to `Peers` ? https://github.com/ava-labs/avalanchego/blob/master/snow/engine/snowman/config.go#L23